src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -1137,6 +1150,43 @@ void dcd_int_handler(uint8_t rhport) {
     // IEPINT bit read-only, clear using DIEPINTn
     handle_ep_irq(rhport, TUSB_DIR_IN);
   }
+
+  // Incomplete isochronous IN transfer interrupt handling.
+  if (gintsts & GINTSTS_IISOIXFR) {
+    dwc2->gintsts = GINTSTS_IISOIXFR;
+    const dwc2_dsts_t dsts = {.value = dwc2->dsts};
+    const uint32_t odd_now = dsts.frame_number & 1u;
+    // Loop over all IN endpoints
+    const uint8_t ep_count = dwc2_ep_count(dwc2);
+    for (uint8_t epnum = 0; epnum < ep_count; epnum++) {
+      dwc2_dep_t* epin = &dwc2->epin[epnum];
+      dwc2_depctl_t depctl = {.value = epin->diepctl};
+      // Read DSTS and DIEPCTLn for all isochronous endpoints. If the current EP is enabled
+      // and the read value of DSTS.SOFFN is the targeted uframe number for this EP, then
+      // this EP has an incomplete transfer.
+      if (depctl.enable && depctl.type == DEPCTL_EPTYPE_ISOCHRONOUS && depctl.dpid_iso_odd == odd_now) {
```

Comment:
is there any chances that 2 ISO is schedule at the same uframe, and 1 is complete and 1 is incomplete and we detect the wrong endpoint ?